### PR TITLE
25244 Showing both cancelled and occurred in the past boxes for cancelled appointments in the past

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR does the following:

- Fixes issue with displaying past and cancelled appointment alerts at the same time.

## Testing done
Unit testing

## Screenshots
Appointment date is in the past
localhost_3001_health-care_schedule-view-va-appointments_appointments_va_f183c1079a800aa475a6752261284bf0
---
localhost_3001_health-care_schedule-view-va-appointments_appointments_va_f183c1079a800aa475a6752261284bf0 (1)

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx [hunk 1]
```diff
@@ -550,6 +550,58 @@ describe('VAOS <ConfirmedAppointmentDetailsPage>', () => {
...
+  it('should display past appointment', async () => {
+    const url = '/va/21cdc6741c00ac67b6cbf6b972d084c1';
+
+    const appointment = getVAAppointmentMock();
+    appointment.attributes = {
+      ...appointment.attributes,
+      clinicId: '308',
+      facilityId: '983',
+      sta6aid: '983GC',
+      startDate: moment().subtract(1, 'day'),
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:562]: This makes it a past appointment.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx [hunk 2]
```diff
@@ -164,6 +164,25 @@ export default function ConfirmedAppointmentDetailsPage() {
     facility: facilityData[locationId],
   });
 
+  let alertBox = null;
+  if (isPastAppointment) {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:168]: I think we'd want to have the canceled alert have first priority here. It's odd to say that an appointment occurred in the past if it was canceled and never occurred.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:168]: Done.

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx [hunk 3]
```diff
@@ -550,6 +550,63 @@ describe('VAOS <ConfirmedAppointmentDetailsPage>', () => {
...
+      await screen.findByText(
+        /Fort Collins VA Clinic canceled this appointment./i,
+      ),
+    ).to.exist;
+
+    // The canceled appointment and past appointment alerts are mutually exclusive
+    // with the canceled appointment status having 1st priority. So, the canceled
+    // appointment alert should display even when the appointment is a past
+    // appointment.
+    expect(screen.queryByText('This appointment occurred in the past.')).not.to
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:606]: Looks like this check needs to be updated with the change in priority

**Review by jbalboni — APPROVED** [review]
LGTM, just have the failing test to fix.

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx (hunk 3 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:606]: Hmm??? The test is passing locally.

Comment (jbalboni) [reply src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:606]: Oh, nevermind, I misread this assertion and then assumed the failed Jenkins build was related.

**Pull request closed without merging**